src/layout.rs

```diff
@@ -393,17 +408,28 @@ impl<const C: usize, const R: usize, const L: usize, T: 'static> Layout<C, R, L,
                 hold,
                 tap,
                 config,
-                ..
+                tap_hold_interval,
             } => {
-                let waiting: WaitingState<T> = WaitingState {
-                    coord,
-                    timeout: *timeout,
-                    delay,
-                    hold,
-                    tap,
-                    config: *config,
-                };
-                self.waiting = Some(waiting);
+                if *tap_hold_interval == 0
+                    || coord != self.tap_hold_tracker.coord
+                    || self.tap_hold_tracker.timeout == 0
+                {
+                    let waiting: WaitingState<T> = WaitingState {
+                        coord,
+                        timeout: *timeout,
+                        delay,
+                        hold,
+                        tap,
+                        config: *config,
+                    };
+                    self.waiting = Some(waiting);
+                    if *tap_hold_interval > 0 {
+                        self.tap_hold_tracker.timeout = *tap_hold_interval;
```

Comment:
I think it's cleaner to do it with the initial press so that `WaitingState` doesn't need to know about `tap_hold_interval`. I don't know how QMK does it, but don't think it matters either way since the user can adjust the timeout to their needs.

Reply:
>  important to activate the countdown only in case the HoldTap resolve to Tap

Implemented in the latest force push by setting timeout to 0 in `waiting_into_hold`